go-app-ussd_ccmdd_wc_address_update.js

```diff
@@ -253,6 +273,354 @@ go.app = (function() {
+        ];
+      } else if (municipality === "Garden Route") {
+        city_choices = [
+          new Choice("Bitou", $("Bitou")),
+          new Choice("George", $("George")),
+          new Choice("Hessequa", $("Hessequa")),
+          new Choice("Kannaland", $("Kannaland")),
+          new Choice("Knysna", $("Knysna")),
+          new Choice("Mosselbay", $("Mosselbay")),
+          new Choice("Oudtshoorn", $("Oudtshoorn"))
+        ];
+      } else if (municipality === "Overberg") {
+        city_choices = [
+          new Choice("Cape Agulhas", $("Cape Agulhas")),
+          new Choice("Overstrand", $("Overstrand")),
+          new Choice("Swellendam", $("Swellendam")),
+          new Choice("Theewaterskloof", $("Theewaterskloof"))
+        ];
+      } else if (municipality === "West Coast") {
+        city_choices = [
+          new Choice("Bergriver", $("Bergriver")),
+          new Choice("Cederberg", $("Cederberg")),
+          new Choice("Matzikama", $("Matzikama")),
+          new Choice("Saldanabay", $("Saldanabay")),
+          new Choice("Swartland", $("Swartland"))
+        ];
+      }
+
+      city_choices.push(
+        new Choice("None of the above", $("None of the above"))
+      );
+
+      return new ChoiceState(name, {
+        question: $("In which city do you stay?"),
+        error: $(
+          "Sorry we don't understand. Please enter the number next to your answer."
+        ),
+        choices: city_choices,
+        next: function(choice) {
+          if (choice.value === "None of the above") {
+            return "state_no_delivery";
+          } else {
+            return "state_suburb";
+          }
+        }
+      });
+    });
+
+    self.add("state_suburb", function(name) {
+      return new FreeText(name, {
+        question: $("Please reply with the name of your suburb."),
+        check: function(content) {
+          var match = content.match(/[a-zA-Z]{2,}/);
+          if (!match) {
+            return $(
+              "Sorry, we don’t understand. Please reply with the name of your " +
+                "suburb, e.g. Woodstock"
+            );
+          }
+        },
+        next: "state_street_name"
+      });
+    });
+
+    self.add("state_street_name", function(name) {
+      return new FreeText(name, {
+        question: $("Please reply with the name of your street."),
+        check: function(content) {
+          var match = content.match(/[a-zA-Z]{2,}/);
+          if (!match) {
+            return $(
+              "Sorry, we don’t understand. Please reply with the name of your " +
+                "street name"
+            );
+          }
+        },
+        next: "state_street_number"
+      });
+    });
+
+    self.add("state_street_number", function(name) {
+      return new FreeText(name, {
+        question: $("Please reply with your house number, e.g. 17."),
+        check: function(content) {
+          var match = content.match(/^(\d{1,7})$/);
```

Comment:
I can see some issues with this being too strict, eg. my number is `32B`

Reply:
hehe, I live at 2A and thought about this